Explorer/Assets/DCL/PerformanceAndDiagnostics/Profiling/IAnalyticsReportProfiler.cs

```diff
@@ -1,14 +1,12 @@
 ﻿namespace DCL.Profiling
 {
-    public interface IAnalyticsReportProfiler : IBudgetProfiler
+    public interface IAnalyticsReportProfiler : IMemoryProfiler
     {
-        long SystemUsedMemoryInBytes { get; }
-
-        long GcUsedMemoryInBytes { get; }
-
-        AnalyticsFrameTimeReport? GetMainThreadFramesNs(int[] percentile);
+        (AnalyticsFrameTimeReport?, long[]) GetMainThreadFramesNs(int[] percentile);
```

Comment:
I think you should avoid modifying the internal output array: you can use a read-only contract instead. If for some reason you do neeed a mutable array I would recommend allocating `Span` on stack on the caller site and pass it as an argument

Reply:
at the end I decided to return already assembled string (as it was required by the client). It was most safe way to not mess up with the re-usage of the `List ` to which I copy